lib/spack/spack/test/concretize.py

```diff
@@ -347,6 +351,10 @@ def test_compiler_flags_differ_identical_compilers(self):
             spec.concretize()
             assert spec.satisfies("cflags=-O2")
 
+    @pytest.mark.skipif(
+        os.environ.get("SPACK_TEST_SOLVER") == "original",
+        reason="Optional compiler propagation isn't deprecated for original concretizer",
+    )
```

Comment:
Why is it necessary to skip this test, while it was not necessary to skip it before? Asking since the logic to deal with flag propagation is separate from variant.

Reply:
This test should've been skipped before since the `==` syntax isn't deprecated

Reply:
The question here was more "Why this was passing before and is skipped now" ? Is it still passing?

Reply:
Thanks for explain. I definitely missed that.